src/workers/personalizedDigestEmail.ts

```diff
@@ -91,45 +96,75 @@ const getPostsTemplateData = ({ posts }: { posts: TemplatePostData[] }) => {
         'email',
         'digest',
       ),
+      post_upvotes: post.upvotes || 0,
+      post_comments: post.comments || 0,
+      post_summary: post.summary,
+      post_read_time: post.readTime,
+      post_views: post.views || 0,
       source_name: post.sourceName,
       source_image: post.sourceImage,
     };
   });
 };
 
-const getEmailVariation = async (
-  variation: number,
-  firstName: string,
-  dayName: string,
-  posts: TemplatePostData[],
-): Promise<Partial<MailDataRequired>> => {
-  const defaultPreview = `Every ${dayName}, we'll send you five posts you haven't read. Each post was carefully picked based on topics you love reading about. Let's get to it!`;
+const getEmailVariation = async ({
+  personalizedDigest,
+  posts,
+  user,
+  feature,
+  currentDate,
+}: {
+  personalizedDigest: UserPersonalizedDigest;
+  posts: TemplatePostData[];
+  user: User;
+  feature: typeof features.personalizedDigest.defaultValue;
+  currentDate: Date;
+}): Promise<Partial<MailDataRequired>> => {
+  const [dayName] = Object.entries(DayOfWeek).find(
+    ([, value]) => value === personalizedDigest.preferredDay,
+  );
+  const userName = user.name?.trim().split(' ')[0] || user.username;
   const data = {
     day_name: dayName,
-    first_name: firstName,
+    first_name: userName,
     posts: getPostsTemplateData({ posts }),
+    date: format(currentDate, 'MMM d, yyyy'),
+    user: {
+      username: user.username,
+      image: user.image,
+      reputation: user.reputation,
+    },
   };
-  if (variation === 2) {
-    return {
-      dynamicTemplateData: {
-        ...data,
-        title: posts[0].title,
-        preview: posts[0].summary || defaultPreview,
-      },
-      from: {
-        email: 'digest@daily.dev',
-        name: 'Weekly Digest',
-      },
-    };
-  }
 
-  return {
+  const mailData = {
+    from: {
+      email: feature.meta.from.email,
+      name: feature.meta.from.name,
+    },
+    to: {
+      email: user.email,
+      name: userName,
+    },
     dynamicTemplateData: {
       ...data,
-      title: `${firstName}, your personal weekly update from daily.dev is ready`,
-      preview: defaultPreview,
+      title: `${userName}, your personal weekly update from daily.dev is ready`,
+      preview: `Every ${dayName}, we'll send you five posts you haven't read. Each post was carefully picked based on topics you love reading about. Let's get to it!`,
```

Comment:
Didn't we want to give them the ability to experiment on that?

Reply:
Yes. Keeping this property for backward compatibility. 

They can experiment by adjusting and using all other variables in templates directly. Both email subject and preview (which is a div inside template body) support variables. This [thread](https://dailydotdev.slack.com/archives/C05G8BHHVBR/p1708006426646369)

Reply:
Hmm should we not still also make this dynamic here?
Looking at upcoming multiple sending days it might be easier for us to support then in sendgrid?

Reply:
I am not sure, would prefer if we use the Sendgrid since it supports it. I will talk to product to see how they feel about usage and then we can adjust if it slows them down by much. WDYT?

Reply:
Sounds like a plan :)